src/components/portfolio/DetailsMainAnalyst.tsx

```diff
@@ -13,25 +22,67 @@ const useStyles = makeStyles(() =>
       display: 'flex',
       flexDirection: 'column',
       width: '100%',
-      // TODO: delete fixed height
-      height: '20rem',
+    },
+    description: {
+      color: palette.primary.contrastText,
+      textAlign: 'center',
+      fontSize: '1.15rem',
     },
   })
 );
 
-// type declarations
+type BarDataType = {
+  [key: number]: string;
+};
+
 type DetailsMainAnalystProps = {
-  // props
+  positions: Position[];
 };
 
-// returns the details page header
-const DetailsMainAnalyst: React.FC<DetailsMainAnalystProps> = () => {
+const DetailsMainAnalyst: React.FC<DetailsMainAnalystProps> = ({
+  positions,
+}) => {
   const classes = useStyles();
+  const theme = useTheme();
+  const { t } = useTranslation();
+
+  // an object that keeps only the data that is needed to display the analyst bar
+  const barData: BarDataType = {};
+
+  if (positions) {
+    Object.values(positions)
+      // sort by total value
+      .sort((a, b) => (a.qty * a.stock.price > b.qty * b.stock.price ? 1 : -1))
+      // take the top 5
+      .slice(positions.length > 5 ? positions.length - 5 : 0)
+      .forEach((p) => {
+        const score = Math.round(p.stock.score * 100);
+
+        if (!barData[score]) {
+          // if there is no entry with this score create one
+          barData[score] = p.stock.name;
+        } else {
+          // otherwise add on to an existing score
+          barData[score] = barData[score].concat(`\n${p.stock.name}`);
+        }
+      });
+  }
```

Comment:
This calculation is performed on every component render. As long as this component has no other props than `positions` and no state, that's fine because the component is only rendered once.

If something is added that causes the component to update, the calculation should be only performed when necessary (e.g. `positions` changed).